crates/router/src/core/payments.rs

```diff
@@ -1609,6 +1609,94 @@ where
     )
 }
 
+#[allow(clippy::too_many_arguments)]
+#[cfg(feature = "v2")]
+pub(crate) async fn payments_create_and_confirm_setup_intent(
+    state: SessionState,
+    req_state: ReqState,
+    merchant_account: domain::MerchantAccount,
+    profile: domain::Profile,
+    key_store: domain::MerchantKeyStore,
+    request: payments_api::PaymentsSetupIntentRequest,
+    mut header_payload: HeaderPayload,
+    platform_merchant_account: Option<domain::MerchantAccount>,
+) -> RouterResponse<payments_api::SetupIntentWrapperResponse> {
+    use actix_http::body::MessageBody;
+    use common_utils::ext_traits::BytesExt;
+    use hyperswitch_domain_models::{
+        payments::{PaymentConfirmData, PaymentIntentData},
+        router_flow_types::{Authorize, SetupMandate},
+    };
+
+    let global_payment_id = id_type::GlobalPaymentId::generate(&state.conf.cell_information.id);
+
+    let payload = payments_api::PaymentsCreateIntentRequest::from(&request);
+
+    let create_intent_response = Box::pin(payments_intent_core::<
```

Comment:
This can be a normal payments_intent_core(PaymentCreateIntent), why we need a separate one?